src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/TasksTest.kt

```diff
@@ -735,6 +736,128 @@ class TasksTest : IntegrationTestBase() {
+              val placementApplicationTasks = mutableMapOf<UserQualification, Task>()
+
+              listOf(
+                UserQualification.WOMENS,
+                UserQualification.ESAP,
+                UserQualification.EMERGENCY,
+                UserQualification.PIPE,
+              ).forEach { qualification ->
+                val (assessment) = `Given an Assessment for Approved Premises`(
+                  allocatedToUser = otherUser,
+                  createdByUser = otherUser,
+                  crn = offenderDetails.otherIds.crn,
+                  requiredQualification = qualification,
+                )
+
+                val (placementRequest, _) = `Given a Placement Request`(
+                  placementRequestAllocatedTo = otherUser,
+                  assessmentAllocatedTo = otherUser,
+                  createdByUser = user,
+                  crn = offenderDetails.otherIds.crn,
+                  requiredQualification = qualification,
+                )
+
+                val placementApplication = `Given a Placement Application`(
+                  createdByUser = user,
+                  allocatedToUser = user,
+                  schema = approvedPremisesPlacementApplicationJsonSchemaEntityFactory.produceAndPersist {
+                    withPermissiveSchema()
+                  },
+                  crn = offenderDetails.otherIds.crn,
+                  submittedAt = OffsetDateTime.now(),
+                  requiredQualification = qualification,
+                )
+
+                assessmentTasks[qualification] = taskTransformer.transformAssessmentToTask(
+                  assessment,
+                  "${offenderDetails.firstName} ${offenderDetails.surname}",
+                )
+                placementRequestTasks[qualification] = taskTransformer.transformPlacementRequestToTask(
+                  placementRequest,
+                  "${offenderDetails.firstName} ${offenderDetails.surname}",
+                )
+                placementApplicationTasks[qualification] = taskTransformer.transformPlacementApplicationToTask(
+                  placementApplication,
+                  "${offenderDetails.firstName} ${offenderDetails.surname}",
+                )
+              }
+
+              tasks = mapOf(
+                TaskType.assessment to assessmentTasks,
+                TaskType.placementRequest to placementRequestTasks,
+                TaskType.placementApplication to placementApplicationTasks,
+              )
+            }
+          }
+        }
+      }
+
+      @ParameterizedTest
+      @CsvSource(
+        "assessment,WOMENS", "assessment,PIPE", "assessment,ESAP", "assessment,EMERGENCY",
+        "placementRequest,WOMENS", "placementRequest,PIPE", "placementRequest,ESAP", "placementRequest,EMERGENCY",
+        "placementApplication,WOMENS", "placementApplication,PIPE", "placementApplication,ESAP", "placementApplication,EMERGENCY",
+      )
```

Comment:
i do love how readable this CsvSource stuff is for describing lots of permutations